instruqt/protecting-your-assets/03-proxying-private-images/assignment.md

```diff
@@ -0,0 +1,251 @@
+        - name: replicated-pull-secret
+```
+
+This lines are only included if the global values is set, in other
+words if the chart was distributed with Replicated. They reference a pull secret
+named `replicated-pull-secret`. Let's create that secret.
+
+
+Create a new file named `imagepullsecret.yaml` in the directory
+`slackernews/templates` to include the secret in your chart.
+
+![Creating the image pull secret](../assets/creating-the-image-pull-secret.png)
+
+If the new file wasn't opened automatically, click on the filename to create
+it. Add the following content to the file. This template uses a guard condition
+to assure it's only included if the injected credentials are available.
+
+```
+{{ if hasKey ((.Values.global).replicated) "dockerconfigjson" }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: replicated-pull-secret
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ .Values.global.replicated.dockerconfigjson }}
+{{ end }}
+```
+
+The editor will automatically save your changes.
+
+Distributing Your Changes
+=========================
+
+Let's release the changes so new installs will use the Replicated proxy
+registry to access the Slackernews and NGINX images.
+
+### Repackaging Your Chart
+
+Since we changed the default values, we need to repackage our Helm chart.
+
+```
+helm package slackernews --destination ./release
+```
+
+You should now have a tarball in directory `release` in your
+home directory named `slackernews-0.5.0.tgz`.
+
+### Creating a New Replicated Release
+
+To make the changes available, we need to release them on the Replicated
+Platform. You'll need to be authenticated to the Replicated Platform to create
+the new release. The simplest way to do that is by specifying an API token in
+your shell environment. The lab setup has created a token for you.
+
+```
+export REPLICATED_API_TOKEN="[[ Instruqt-Var key="REPLICATED_API_TOKEN" hostname="shell" ]]"
+```
+
+The `replicated` command also needs to know which application to work with. You
+can set it with a command flag or with an environment variable. Let's set the
+variable to save some typing.
+
+```
+export REPLICATED_APP="[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]"
+```
+
+To create a new release, run the following command.
+
+```
+replicated release create --promote Unstable --chart ./release/slackernews-0.5.0.tgz --version 0.5.0  \
+  --release-notes "Uses the Replicated proxy registry to pull images"
+```
+
+This will not only create the release but push it to the `Unstable` release
+channel on the platform. It will also output a sequence number, which should be
+`5` (if it's not, use the number in the output for the upcoming commands).
+We're going to release our changes on three different release channels in this
+lab:
+
+* `Unstable`: use for internal releases that aren't ready for customers
+* `Beta`: for releases that are ready for beta testing
+* `Stable`: the general availability channel for all releases
+
```

Comment:
I also got a comment from the CLI that I was creating a helm only release. Might be nice to let people know this same release can be extended to KOTS/Embedded Cluster this isn't work that is only for helm cli.

Reply:
Good point.